src/app/core/shell/toolbar/toolbar.component.ts

```diff
@@ -80,9 +80,9 @@ export class ToolbarComponent implements OnInit {
   ngDoCheck() {
     this.userData = this.authenticationService.getCredentials();
     
-    if(!this.hasChecked && this.userData?.permissions.length > 0) {
-      let hasAnyPermission = this.allowedPermissions.filter(item => this.userData.permissions.includes(item)).length > 0;
-      if(hasAnyPermission) {
+    if(!this.hasChecked && this.userData?.roles.length > 0) {
+      let isEligible = this.userData.roles.filter(item => this.allowedRoles.includes(item.name.toLowerCase())).length > 0;
```

Comment:
I think this operation is `O(m x n)`  where `m` is the number of items in `allowedRoles`, and `n` is the number of items in `userData.roles`

Consider making `allowedRoles` a Set, to improve the complexity to O(n).